test/types/hooks.test-d.ts

```diff
@@ -256,40 +256,82 @@ type CustomContextConfig = FastifyContextConfig & {
 server.route<RouteGenericInterface, CustomContextConfig>({
   method: 'GET',
   url: '/',
-  handler: () => {},
-  onRequest: (request, reply) => {
+  handler: () => { },
+  onRequest: (request, reply, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  preParsing: (request, reply) => {
+  preParsing: (request, reply, payload, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  preValidation: (request, reply) => {
+  preValidation: (request, reply, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  preHandler: (request, reply) => {
+  preHandler: (request, reply, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  preSerialization: (request, reply) => {
+  preSerialization: (request, reply, payload, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  onSend: (request, reply) => {
+  onSend: (request, reply, payload, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  onResponse: (request, reply) => {
+  onResponse: (request, reply, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  onTimeout: (request, reply) => {
+  onTimeout: (request, reply, done) => {
     expectType<CustomContextConfig>(request.context.config)
     expectType<CustomContextConfig>(reply.context.config)
   },
-  onError: (request, reply) => {
+  onError: (request, reply, error, done) => {
+    expectType<CustomContextConfig>(request.context.config)
+    expectType<CustomContextConfig>(reply.context.config)
+  }
+})
+
+server.route<RouteGenericInterface, CustomContextConfig>({
+  method: 'GET',
+  url: '/',
+  handler: () => { },
+  onRequest: async (request, reply, done) => {
```

Comment:
This is incorrect. When `async` is specified there should be no `done` callback.

Reply:
Jees, why is the guy adding done or error payload in the code even when they are not being used in the function body?

If possible, can I correct this one?

Reply:
Sure thing, send a fresh PR.